src/Raven.Server/Documents/Subscriptions/SubscriptionStorage.cs

```diff
@@ -160,6 +161,30 @@ public async Task AcknowledgeBatchProcessed(long subscriptionId, string name, st
                 DocumentsToResend = docsToResend
             };
 
+            var state = _serverStore.Engine.CurrentState;
+            if (command.ChangeVector == nameof(Constants.Documents.SubscriptionChangeVectorSpecialStates.DoNotChange) &&
+                (state == RachisState.Leader || state == RachisState.Follower))
```

Comment:
I'm checking the state of the subscription on the node that is handling it instead of sending it as a command to the leader.
@garayx @karmeli87